src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -477,22 +477,22 @@ private function needsCdataWrapping($val)
      */
     private function selectNodeType(\DOMNode $node, $val)
     {
-        if (is_array($val)) {
+        if (\is_array($val)) {
             return $this->buildXml($node, $val);
         } elseif ($val instanceof \SimpleXMLElement) {
             $child = $this->dom->importNode(dom_import_simplexml($val), true);
             $node->appendChild($child);
         } elseif ($val instanceof \Traversable) {
             $this->buildXml($node, $val);
-        } elseif (is_object($val)) {
+        } elseif (\is_object($val)) {
             return $this->selectNodeType($node, $this->serializer->normalize($val, $this->format, $this->context));
         } elseif (is_numeric($val)) {
             return $this->appendText($node, (string) $val);
-        } elseif (is_string($val) && $this->needsCdataWrapping($val)) {
+        } elseif (($isString = \is_string($val)) && $this->needsCdataWrapping($val)) {
             return $this->appendCData($node, $val);
-        } elseif (is_string($val)) {
+        } elseif ($isString) {
```

Comment:
i'm not sure using a variable is faster at all (same below)

Reply:
You mean the call is cheap and it's better to duplicate the call to `is_string`?

Reply:
yep: the call is not an call anymore, it's an opcode

Reply:
This could be written like this maybe:
```php
} elseif (\is_string($val)) {
    if ($this->needsCdataWrapping($val)) {
        return $this->appendCData($node, $val);
    } else {
        return $this->appendText($node, $val);
    }
} elseif {
```

Reply:
this wouldn't bring anything, the check if almost free